### ISSUE TITLE
problems in reproducing multitask results

### ISSUE THREAD

**Issue by ShawnKS** [deepchem/deepchem#2879]
## ❓ Questions & Help
Hey guys, I just tried the multitask learning scripts in this repo. However, it found that the MTL results are far different from the results reported in this paper
> Wu, Zhenqin, et al. "MoleculeNet: a benchmark for molecular machine learning." Chemical science 9.2 (2018): 513-530.

For example, the Clintox's AUROC results is 0.59, 0.83 in our trial, compared to the 0.71\~0.72 and 0.77\~0.78 as illustrated in the paper's Figure 9.
Since the dataset split in a 'random' manner, it's also strange to see such performance gap among results between test set and valid set. (And it seems there's no early stopping strategy used)
Here's the code I used for Clintox MTL experiment(https://github.com/deepchem/deepchem/blob/master/examples/clintox/clintox_fcnet.py)

```python
import numpy as np
import deepchem as dc
from deepchem.molnet import load_clintox
np.random.seed(123)

tasks, datasets, transformers = load_clintox(split='random')
train_dataset, valid_dataset, test_dataset = datasets
n_features = train_dataset.get_data_shape()[0]
metric = dc.metrics.Metric(dc.metrics.roc_auc_score, np.mean)
model = dc.models.MultitaskClassifier(
    len(tasks),
    n_features,
    layer_sizes=[512],
    dropouts=[.25],
    learning_rate=0.001,
    batch_size=64)
model.fit(train_dataset)
print("Evaluating model")
train_scores = model.evaluate(train_dataset, [metric], transformers)
valid_scores = model.evaluate(valid_dataset, [metric], transformers)
test_scores = model.evaluate(test_dataset, [metric], transformers)

print("Train scores")
print(train_scores)

print("Validation scores")
print(valid_scores)

print("Test scores")
print(test_scores)
```

![_3ZPORA@L7ZKVWT~J(TYC`L](https://user-images.githubusercontent.com/26742152/156297283-43d1a45f-0eef-4368-a5c8-61a400e7f980.png)
![)6N8@RTGEL(QWEM%GUV)J8F](https://user-images.githubusercontent.com/26742152/156297355-efd2ca69-6c59-4c5d-b8d1-991997b4ade2.png)

**Comment by ShawnKS** [comment]
And the auprc results on muv is far different from the results in paper
![@7A478B9OUA)%AXBY$NHV00](https://user-images.githubusercontent.com/26742152/156298465-e55cb823-dd72-4187-83be-e9e355fc7ae8.png)
![image](https://user-images.githubusercontent.com/26742152/156298535-62383e05-f029-4f44-9bc7-a42666f00d82.png)
```python

import numpy as np
import deepchem as dc
from deepchem.molnet import load_muv

# Only for debug!
np.random.seed(123)

# Load clintox dataset
tasks, datasets, transformers = load_muv(split='random')
train_dataset, valid_dataset, test_dataset = datasets
n_features = train_dataset.get_data_shape()[0]

# Fit models
# metric = dc.metrics.Metric(dc.metrics.roc_auc_score, np.mean)
metric = dc.metrics.Metric(dc.metrics.prc_auc_score, np.mean)
'''dc.models.MultitaskClassifier 
https://github.com/deepchem/deepchem/blob/master/deepchem/models/torch_models/torch_model.py
https://github.com/deepchem/deepchem/blob/master/deepchem/models/fcnet.py
'''
model = dc.models.MultitaskClassifier(
    len(tasks),
    n_features,
    layer_sizes=[512],
    dropouts=[.25],
    learning_rate=0.001,
    batch_size=64)
# Fit trained model
model.fit(train_dataset)
'''
function fit() 
https://github.com/deepchem/deepchem/blob/master/deepchem/models/torch_models/torch_model.py
line 285
'''
print("Evaluating model")
train_scores = model.evaluate(train_dataset, [metric], transformers)
valid_scores = model.evaluate(valid_dataset, [metric], transformers)
test_scores = model.evaluate(test_dataset, [metric], transformers)

print("Train scores")
print(train_scores)

print("Validation scores")
print(valid_scores)

print("Test scores")
print(test_scores)


```


p.s.: The nan value occurs due to the 'random' split.

**Comment by rbharath** [comment]
The original numbers were run on a much older version of TensorFlow than is currently being used. I believe also that `MultitaskClassifier` has now been ported to PyTorch in more recent versions of DeepChem, meaning that the backend is now entirely different. In general, exactly matching numbers for benchmarks across different versions of numerical backends is very challenging and multiple teams have reported issues here. My suspicion is if you retune hyperparameters for the latest version, you will be able to match or exceed the original numbers in the MoleculeNet paper.

**Comment by rbharath** [comment]
Going to close this issue. Please feel free to re-open if you have more questions!
